x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -684,4 +771,35 @@ protected LogicalPlan rule(Limit plan) {
             return p;
         }
     }
+
+    public static class ReplaceRegexMatch extends OptimizerRules.OptimizerExpressionRule<RegexMatch<?>> {
```

Comment:
No need to copy the class - extend ReplaceRegexMatch in QL and override regextToEquals to return the ESQL Equals class similar to EQL.